### DISCUSSION TITLE
tree-sitter generate: generating test-wrong code

### DISCUSSION THREAD

**Discussion by ArmProg-San** [tree-sitter/tree-sitter#2675] [Q&A] [answered]
Hello, I ran the tree-sitter test code, and it passed all tests. However, after running tree-sitter generate for tree-sitter-php, when I ran the test using tree-sitter test, the code couldn't pass several tests.
`tree-sitter test
  statements:
    ✓ If statements
    ✓ Alternative if statements
    ✓ Wordpress colon blocks
    ✓ While statements
    ✓ Alternative while statements
    ✓ For statements
    ✓ Switch statements
    ✓ Alternative switch statements
    ✓ Include statement
    ✓ Do-while statements
    ✓ Try statements
    ✓ Foreach statements
    ✓ Case insensitive keywords
    ✓ Accessing Constants
  expressions:
    ✓ Dynamic variable names
    ✓ Exponentiation
    ✓ Reserved Identifiers as Names
    ✓ Unary op with assignment expressions
    ✓ Capitalized logical operators
    ✓ Cast expressions in assignments
    ✓ Cast include
    ✓ Reserved words as function calls
    ✓ Scoped self call expressions
    ✓ Symmetric array destructuring
    ✓ Precedence of assignment, `or`, `xor`, and `and`
    ✓ Associativity of conditional
    ✓ Associativity of null-coalescence
    ✓ Associativity of negation
    ✓ Augmented assignment
    ✓ Nested assignemnts
    ✓ All binary op precedence levels
    ✓ Concatenation precedence
    ✓ Arrays
    ✓ Anonymous functions
    ✓ Throw expressions
    ✓ Nullsafe operator
    ✓ First class callable syntax
    ✓ Match expressions
    ✓ Arrow functions
    ✓ Functions with named arguments
    ✓ Precedence between concatenation and shift
    ✓ References
    ✓ Empty match expressions
  literals:
    ✓ Booleans
    ✓ Floats
    ✓ Integers
    ✓ Testing string scanner confirmance
    ✓ Shell command
  execution_operator:
    ✓ Quotes in backticks
    ✓ Escape sequences in backticks
    ✓ Variable interpolation in backticks
    ✓ Member access expression in backticks
    ✓ Array subscript expression in backticks
    ✓ Complex interpolation in backticks
    ✓ Nesting of expression in backticks
    ✓ Nested escaped backticks
    ✓ Comment in backticks
  string:
    ✓ Complex: Variable access
    ✓ Complex: Disallow space between { and $
    ✓ Complex: PHP documentation tests
    ✓ Simple: Variable access
    ✓ Simple: Member and array access
    ✓ Corner cases
    ✓ Single quoted
    ✓ Bug: #113
    ✓ Heredocs
    ✓ Nowdocs
    ✓ Unicode escape sequences
  class:
    ✓ Abstract class
    ✓ Anonymous classes
    ✓ Conditional class definition
    ✓ Class constant modifiers
    ✓ Final class
    ✓ Implicitly public properties and methods
    ✓ Property Types
    ✓ Constructor Property Promotion
    ✓ Readonly properties
    ✓ Constructor property promotion with readonly modifier
    ✓ Readonly class
    ✓ Constants in trait
  types:
    ✓ Type names
    ✓ Primitive types
    ✓ Intersection type
    ✓ Optional types
    ✓ Union types
    ✓ Mixed type
    ✓ Static type
    ✓ Null type
    ✓ False type
    ✓ True type
  bugs:
    ✓ #131: Parse error when using self as constant
    ✓ #133: Incorrect precedence for error supression operator
  interpolation:
    ✓ no interpolated text
    ✓ interpolated text at beginning
    ✓ interpolated text at end
    ✓ interpolated text in middle
    ✓ short open tag: On
    ✓ short open tag: Off
    ✓ Single line php comment
    ✓ Singel line comment without any content
    ✓ Closing tags before the first PHP tag
  declarations:
    ✓ Interface declarations
    ✓ Use declarations
    ✓ Namespace names in namespaces
    ✓ Class declarations
    ✓ Class declarations with base classes
    ✓ Function parameters
    ✓ Functions with default parameters
    ✓ Static variables in functions
    ✓ Defining Constants
    ✓ Attributes
    ✓ Enums
syntax highlighting:
  ✓ literals.php (7 assertions)
  ✓ types.php (3 assertions)
  ✓ keywords.php (18 assertions)`
  
after generate:
`
  statements:
    ✓ If statements
    ✓ Alternative if statements
    ✓ Wordpress colon blocks
    ✓ While statements
    ✓ Alternative while statements
    ✓ For statements
    ✓ Switch statements
    ✓ Alternative switch statements
    ✓ Include statement
    ✓ Do-while statements
    ✓ Try statements
    ✓ Foreach statements
    ✗ Case insensitive keywords
    ✓ Accessing Constants
  expressions:
    ✓ Dynamic variable names
    ✓ Exponentiation
    ✓ Reserved Identifiers as Names
    ✓ Unary op with assignment expressions
    ✗ Capitalized logical operators
    ✓ Cast expressions in assignments
    ✓ Cast include
    ✓ Reserved words as function calls
    ✓ Scoped self call expressions
    ✓ Symmetric array destructuring
    ✓ Precedence of assignment, `or`, `xor`, and `and`
    ✓ Associativity of conditional
    ✗ Associativity of null-coalescence
    ✓ Associativity of negation
    ✓ Augmented assignment
    ✓ Nested assignemnts
    ✓ All binary op precedence levels
    ✓ Concatenation precedence
    ✓ Arrays
    ✓ Anonymous functions
    ✓ Throw expressions
    ✓ Nullsafe operator
    ✓ First class callable syntax
    ✓ Match expressions
    ✓ Arrow functions
    ✓ Functions with named arguments
    ✓ Precedence between concatenation and shift
    ✓ References
    ✓ Empty match expressions
  literals:
    ✓ Booleans
    ✓ Floats
    ✓ Integers
    ✓ Testing string scanner confirmance
    ✓ Shell command
  execution_operator:
    ✓ Quotes in backticks
    ✓ Escape sequences in backticks
    ✓ Variable interpolation in backticks
    ✓ Member access expression in backticks
    ✓ Array subscript expression in backticks
    ✓ Complex interpolation in backticks
    ✓ Nesting of expression in backticks
    ✓ Nested escaped backticks
    ✓ Comment in backticks
  string:
    ✓ Complex: Variable access
    ✓ Complex: Disallow space between { and $
    ✓ Complex: PHP documentation tests
    ✓ Simple: Variable access
    ✓ Simple: Member and array access
    ✓ Corner cases
    ✓ Single quoted
    ✓ Bug: #113
    ✓ Heredocs
    ✓ Nowdocs
    ✓ Unicode escape sequences
  class:
    ✓ Abstract class
    ✓ Anonymous classes
    ✓ Conditional class definition
    ✓ Class constant modifiers
    ✓ Final class
    ✓ Implicitly public properties and methods
    ✓ Property Types
    ✓ Constructor Property Promotion
    ✗ Readonly properties
    ✓ Constructor property promotion with readonly modifier
    ✓ Readonly class
    ✓ Constants in trait
  types:
    ✓ Type names
    ✓ Primitive types
    ✓ Intersection type
    ✓ Optional types
    ✓ Union types
    ✓ Mixed type
    ✓ Static type
    ✓ Null type
    ✓ False type
    ✓ True type
  bugs:
    ✓ #131: Parse error when using self as constant
    ✓ #133: Incorrect precedence for error supression operator
  interpolation:
    ✓ no interpolated text
    ✓ interpolated text at beginning
    ✓ interpolated text at end
    ✓ interpolated text in middle
    ✓ short open tag: On
    ✓ short open tag: Off
    ✓ Single line php comment
    ✓ Singel line comment without any content
    ✓ Closing tags before the first PHP tag
  declarations:
    ✓ Interface declarations
    ✓ Use declarations
    ✓ Namespace names in namespaces
    ✓ Class declarations
    ✓ Class declarations with base classes
    ✓ Function parameters
    ✓ Functions with default parameters
    ✓ Static variables in functions
    ✓ Defining Constants
    ✓ Attributes
    ✓ Enums

4 failures:

expected / actual

  1. Case insensitive keywords:

    (program
      (php_tag)
      (expression_statement
        (subscript_expression
          (function_call_expression
            (name)
            (arguments
              (argument
                (dynamic_variable_name
                  (ERROR
                    (name)
                    (name))
      (foreach_statement
        (variable_name
          (name))
        (variable_name
          (name))
        (compound_statement
          (do_statement
            (compound_statement
              (if_statement
                (parenthesized_expression
                  (variable_name
                    (name))))))
          (function_call_expression
            (subscript_expression
              (name)
              (subscript_expression
                (subscript_expression
                  (function_call_expression
                    (name)
                    (arguments
                      (argument
                        (variable_name
                          (name)))))
                  (function_call_expression
                    (name)
                    (arguments))
                  (ERROR))
                (ERROR
                  (name))
                (function_call_expression
                  (name)
                  (arguments))
                (ERROR)))
            (ERROR
              (name))
            (arguments
              (argument
                (variable_name
                  (name)))))
          (ERROR))
        (MISSING ";")))
                    (name)))
                (compound_statement
                  (expression_statement
                    (function_call_expression
                      (name)
                      (arguments))))
                (else_clause
                  (compound_statement
                    (expression_statement
                      (function_call_expression
                        (name)
                        (arguments)))))))
            (parenthesized_expression
              (variable_name
                (name)))))))

  2. Capitalized logical operators:

    (program
      (php_tag)
      (expression_statement
        (function_call_expression
          (name)
          (arguments
            (ERROR
              (name))
            (argument
              (function_call_expression
                (function_call_expression
                  (name)
                  (arguments))
                (ERROR
                  (name)
                  (name))
                (arguments)))
            (MISSING ")))))
        (binary_expression
          (function_call_expression
            (name)
            (arguments))
          (binary_expression
            (function_call_expression
              (name)
              (arguments))
            (function_call_expression
              (name)
              (arguments))))))

  3. Associativity of null-coalescence:

    (program
      (php_tag)
      (expression_statement
        (binary_expression
          (name)
          (null)
          (binary_expression
            (name)
            (null)
            (integer))))
      (text_interpolation))

  4. Readonly properties:

    (program
      (php_tag)
      (class_declaration
        (name)
        (declaration_list
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (named_type
                (name)))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (named_type
                (name)))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (named_type
                (name)))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (named_type
                (name)))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (visibility_modifier)
            (readonly_modifier)
            (union_type
              (named_type
                (name)))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (visibility_modifier)
            (union_type
              (named_type
                (name)))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (union_type
              (primitive_type))
            (ERROR
              (property_element
                (variable_name
                  (name)))
              (name)
              (property_element
                (variable_name
                  (name)))
              (name))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (union_type
              (primitive_type))
            (property_element
              (variable_name
                (name))))
          (property_declaration
            (readonly_modifier)
            (union_type
              (named_type
                (name)))
            (property_element
              (variable_name
                (name)))))))`

I also attempted `npm install` and then `npm run build` and `npm run test`, but I obtained the same result.

System Information:
- Ubuntu 22.04.3 LTS
- Node.js version: v20.8.0
- npm version: 10.1.0
- tree-sitter version: 0.20.8
- tree-sitter-php version: 0.19.0
- nan: 2.18.0

Please advise on what I might have missed or any steps I can take to resolve these test failures.

**Comment by sogaiu** [comment, marked as answer]
I don't get the same results.

All tests pass (via `tree-sitter test`) here whether `tree-sitter generate` is done first or not.

I removed `php.so` from under `~/.cache/tree-sitter/lib/` to make sure the processing was using a freshly created `.so` file.

tree-sitter (2cbfcbc35a7d245968f07ef5e9dee1a3ea574fee)
tree-sitter-php (a05c6112a1dfdd9e586cb275700931e68d3c7c85)

Also, may be it makes sense to check with [tree-sitter-php](https://github.com/tree-sitter/tree-sitter-php) repository folks first?

**Reply by sogaiu** [reply]
Possibly related https://github.com/tree-sitter/tree-sitter/issues/2517
